lib/usd/ui/cmds/USDImportDialogCmd.cpp

```diff
@@ -0,0 +1,122 @@
+//
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "USDImportDialogCmd.h"
+#include "views/USDImportDialog.h"
+
+#include <mayaUsd/fileio/importData.h>
+
+#include <maya/MStatus.h>
+#include <maya/MSyntax.h>
+#include <maya/MArgParser.h>
+#include <maya/MString.h>
+#include <maya/MStringArray.h>
+#include <maya/MFileObject.h>
+#include <maya/MQtUtil.h>
+
+// This is added to prevent multiple definitions of the MApiVersion string.
+#define MNoVersionString
+#include <maya/MFnPlugin.h>
+
+MAYAUSD_NS_DEF {
+
+const MString USDImportDialogCmd::fsName("usdImportDialog");
+
+namespace {
+
+constexpr auto kClearDataFlag = "-cd";
+constexpr auto kClearDataFlagLong = "-clearData";
+
+}
+
+USDImportDialogCmd::USDImportDialogCmd()
+{
+}
+
+USDImportDialogCmd::~USDImportDialogCmd()
+{
+}
+
+/*static*/
+MStatus USDImportDialogCmd::initialize(MFnPlugin& plugin)
+{
+	return plugin.registerCommand(
+		fsName,
+		USDImportDialogCmd::creator,
+		USDImportDialogCmd::createSyntax);
+}
+
+/*static*/
+MStatus USDImportDialogCmd::finalize(MFnPlugin& plugin)
+{
+	return plugin.deregisterCommand(fsName);
+}
+
+/*static*/
+void* USDImportDialogCmd::creator()
+{
+	return new USDImportDialogCmd();
+}
+
+MStatus USDImportDialogCmd::doIt(const MArgList& args)
+{
+	MStatus st;
+	MArgParser argData(syntax(), args, &st);
+
+	if(argData.isFlagSet(kClearDataFlag))
```

Comment:
The clear data flag is called by the import dialog when you click on a new file.